File: spring-cloud-integrated-demo/account-service/account-service-api/src/main/java/com.github.jccode.springcloud.integrateddemo.account/api/AccountAPIFallback.java
package com.github.jccode.springcloud.integrateddemo.account.api;

import com.github.jccode.springbootsample.core.data.rest.RestResult;
import org.springframework.stereotype.Component;

import static com.github.jccode.springbootsample.core.utils.CommonUtil.fail;

@Component
class AccountAPIFallback implements AccountAPI {

    @Override
    public RestResult<Integer> getBalance(Integer userId) {
        return fail("Get user balance fail.", 0);
    }
}

File: spring-cloud-integrated-demo/account-service/account-service-api/src/main/scala/com/github/jccode/springcloud/integrateddemo/account/api/AccountApiFallback.scala
package com.github.jccode.springcloud.integrateddemo.account.api

import com.github.jccode.springbootsample.core.data.RestResult
import com.github.jccode.springbootsample.core.data.RestResult.RestSuccess
import org.springframework.stereotype.Component

@Component
class AccountApiFallback extends AccountApi {
  override def getBalance(userId: Integer): RestSuccess[Integer] = RestResult.fail("Get user balance failed")
}
